opendevin/sandbox/docker/ssh_box.py

```diff
@@ -15,6 +15,8 @@
 from opendevin.sandbox.sandbox import Sandbox
 from opendevin.sandbox.process import Process
 from opendevin.sandbox.docker.process import DockerProcess
+from opendevin.sandbox.plugins.mixin import PluginMixin
+from opendevin.sandbox.plugins.jupyter import JupyterRequirement
```

Comment:
Rather than importing the requirements one by one in the sandboxes, I'm imagining something like:
* CodeActAgent defines `CodeActAgent.sandbox_plugins=['jupyter']`
* When the AgentController starts up, it sees this, and passes a `JupyterRequirement` to whatever sandbox is being used (either in the `__init__` or in another method)

Reply:
done - i set `sandbox_plugins=[JupyterRequirement()]` to avoid some code that maps 'jupyter' to `JupyterRequirement`, not sure which one is the best way though.